feat: badger day update

381e35 opened the pull request:
PR #174 rebuilds most of the foundation of the web-app to operate on the framework needed to leave the MVP phase. With this, WebSockets become integrated, followed by clean-up in the supporting areas.

We have had to handle a bit of technical debt in the beginning; thus, this PR rebuilds every piece of the app. While we knew that we would one day want WebSockets, we did not expect to change our approach in such a severe manner.

To support this, the pieces are being developed in parallel to make the:
- API
- Frontend
- Smart Contract

All work together without compromise. Avoiding the typical downfall of traditional web3 app development, the on-chain team is the same as the off-chain; thus, we can secure benefits built into the product's core.

Moving forward, the technical overhead and developer onboarding are significantly lower due to the approachability and general architecture of the API and the consuming routes.

Once the app experience has been updated, the brand will also be updated in this same PR.

- [x] Resolved by #207 with the deprecation of auth: Authentication is nice and smooth.
- [x] #197
- [x] #198
- [x] c2b041c40e1d9e096ffd7c232213f94fc4dbf043 : Implement API support for balances and Badges as source.
- [x] #211 Frontend consumption of #198 & 658c7f4b363a5aa93c720aae2d5a574272856f30.
- [x] #206 Use the Badge websocket source as the data source for Badges.
     - [x] This will require slight data shifting up one level to drill down.
- [x] Badge managers (just people no configuration) on the frontend.
     
**Nice to haves:**

- [x] bba074633e8fe6f6d95bd114d42d98aa9dc50d12 : Refactor background jobs to be balancing / non-critical functionality.
    - [x] Make sure that we are constantly updated with balances.
    - [x] Refresh the static metadata of the Organization every so often.
    - [x] Fill in IPFS metadata asynchronously instead of blocking.

nftchance commented on frontend/src/pages/Badge/BadgeForm.js:
```diff
@@ -17,320 +18,120 @@ import {
-    );
+    const { authenticatedAddress, chain, organization, badge } = useUser({ orgId, badgeId });
+
+    const [ obj, dispatchObj ] = useReducer(FormReducer, badge || initialBadgeForm);
```
I need a crash course on this.

nftchance commented on frontend/src/hooks/useLogout.js:
```diff
@@ -11,10 +11,12 @@ const useLogout = () => {
         // document.cookie = 'csrftoken=; Path=/; Expires=Sat, 01 Jan 2000 00:00:001 GMT;';
-
+        
+        // TODO: We cannot clear the sessionid cookie because it is an HTTP_ONLY cookie, so we need to find 
```
The answer is not to remove things. Yes, I am aware :)

nftchance replied:
The above commented-out line has the answer already ^

nftchance commented on frontend/src/contexts/AuthenticationContext.js:
```diff
@@ -25,7 +25,7 @@ const AuthenticationContextProvider = ({ children }) => {
 
     useEffect(() => {
-        if (isWrongNetwork) switchNetwork(primaryChain.id)
+        if (isWrongNetwork) switchNetwork?.(primaryChain.id)
```
Proper way of doing this is:

`if (isAuthenticated && isWrongNetwork)`

masonchain replied:
Tried that, still was crashing with "switchNetwork is not a function"

nftchance replied:
Yeah I mean I'm no Wagmi expert I don't know the actual library, I just meant please let's ignore the awful syntax and approach that they suggest; it is very bad

nftchance commented:
- [ ] Indexer running locally, so we can use that for the backfill data source.
- [ ] Authentication is nice and smooth.
- [ ] Need to get all the forms connected again.
- [ ] Finalize the indexer to index Managers and Hooks.
- [ ] Use the Badge websocket source as the data source for Badges.
     - [ ] This will require slight data shifting up one level to drill down.

masonchain committed:
Merge pull request #208 from FlipsideCrypto/fix/generated-images

fix: generated images and color cards

masonchain committed:
Merge branch 'feat-badger-day' into feat/multicall

masonchain committed:
Delete useSetManagers.js

masonchain committed:
fix: gas report

masonchain committed:
fix: remove badge danger zone ref

masonchain committed:
Merge branch 'feat/multicall' of https://github.com/FlipsideCrypto/badger into feat/multicall

masonchain committed:
fix: gas report

381e35 committed:
fix: get_ethereum_address bug

381e35 committed:
fix: better badge image display

381e35 committed:
fix: style simplifcation

masonchain committed:
Merge branch 'feat-badger-day' into feat/multicall

masonchain committed:
fix: gas report

381e35 committed:
fix: continued style cleanup

381e35 committed:
fix: move delay a card down

381e35 committed:
fix: serializers

masonchain committed:
fix: display holders

masonchain committed:
Merge branch 'feat/multicall' of https://github.com/FlipsideCrypto/badger into feat/multicall

masonchain committed:
fix: gas report

381e35 committed:
fix: force amount

masonchain committed:
Merge branch 'feat-badger-day' into feat/multicall

masonchain committed:
fix: gas report

masonchain committed:
Merge pull request #209 from FlipsideCrypto/feat/multicall

feat: transfer bound & multicall

381e35 committed:
fix: serializer bug + indexer balance bugs

381e35 committed:
feat: add config...() to loader

381e35 committed:
fix: api bugs

381e35 committed:
fix: better kerning on badge art

381e35 committed:
fix: docs

381e35 committed:
fix: hour bound seed

381e35 committed:
feat: debounce image generation

nftchance committed:
fix: gas report

381e35 committed:
fix: badge table images

381e35 committed:
Merge branch 'fix/spaces-in-badge-name' of https://github.com/FlipsideCrypto/badger into fix/spaces-in-badge-name

nftchance committed:
fix: gas report

381e35 committed:
fix: preview__container

381e35 committed:
Merge branch 'fix/spaces-in-badge-name' of https://github.com/FlipsideCrypto/badger into fix/spaces-in-badge-name

nftchance committed:
fix: gas report

masonchain committed:
wip: holder table multicall mint and revoke

381e35 committed:
fix: stable org card image

nftchance committed:
fix: gas report

381e35 committed:
fix: gateway import

381e35 committed:
Merge branch 'fix/spaces-in-badge-name' of https://github.com/FlipsideCrypto/badger into fix/spaces-in-badge-name

nftchance committed:
fix: gas report

381e35 committed:
fix: deprecate poppins use

381e35 committed:
Merge branch 'fix/spaces-in-badge-name' of https://github.com/FlipsideCrypto/badger into fix/spaces-in-badge-name

381e35 committed:
fix: QOL + brand color and styling

nftchance committed:
fix: gas report

381e35 committed:
fix: fixing badge description width

nftchance committed:
fix: gas report

381e35 committed:
fix: spaces in badge name + styling updates (#210)

This includes a weird bunch of bug fixes:

- [x] Fixes not being able to have spaces in a badge name.
- [x] Fixes the flashing of colors in the image when editing a badge name.
- [x] Fixes the spamming the database by the form for images.
- [x] Fixes the basing of colors to date-hour-address meaning Badges created within the same hour by the same Organization, will have shared color coding.
    - This is kind of like having Seasonal Badges where if someone makes a ton of Badges in a short span, they will visually belong together.
- [x] Fixes the overflow of some Badge names not fitting on their line.
- [x] Fixes the lack of '-' when a word is truncated to another line in the Badge art.
- [x] Fixes `BadgeTable` to use `ImageLoader`.
- [x] Fixes the empty display of Badge art.
- [x] Fixes image scooting around when it has not yet loaded.
- [x] Fixes the font used in the Badge art to conform across devices.
- [x] Fixes colors used in the brand.
- [x] Fixes buttons not having `active`,`hover`,`focus` states.
- [x] Fixes missing primary clickable components `hover` effect.
- [x] Fixes buttons that are hard to discern.
- [x] Fixes primary call to action button not filling form size.
- [x] Fixes banner size.
- [x] Fixes conformity of colors used across the entire app.

masonchain committed:
feat: multicall mint and revoking

masonchain committed:
fix: make code more parseable

381e35 committed:
feat: user creation ens integration

381e35 committed:
fix: save() ens fields

381e35 committed:
Merge branch 'feat-badger-day' into feat/resolve-all-ens

nftchance committed:
fix: gas report

381e35 committed:
feat: resolve ens name and avatar for all useres (#212)

masonchain committed:
feat: init setmanager hook

masonchain committed:
fix: all contract hooks into own file

381e35 committed:
fix: holder table multicall mint and revoking (#211)

masonchain committed:
fix: remove head row constants

masonchain committed:
fix: setManagers bug

masonchain committed:
fix: undefined error

masonchain committed:
fix: small bug fix

masonchain committed:
fix: catching null arguments

masonchain committed:
fix: badge form bug

masonchain committed:
fix: badge form bug

masonchain committed:
feat: badge form danger zone

masonchain committed:
Merge branch 'feat/managers' into feat/danger-zone

masonchain committed:
feat: transfer ownership

masonchain committed:
fix: gas report

381e35 committed:
feat: better architecture

nftchance committed:
fix: gas report

381e35 committed:
fix: bytes decoding + state bugs

nftchance committed:
fix: gas report

381e35 committed:
fix: bug clean up

381e35 committed:
fix: hex manager and hook keys

381e35 committed:
fix: canManage permissions

381e35 committed:
Merge branch 'feat-badger-day' of https://github.com/FlipsideCrypto/badger into feat/user-context

nftchance committed:
fix: gas report

381e35 committed:
fix: loose bugs

nftchance committed:
fix: gas report

masonchain committed:
Merge pull request #220 from FlipsideCrypto/feat/user-context

feat: better user architecture

masonchain committed:
fix: gas report

masonchain committed:
Merge feat/managers into fix/v6-consolidation

masonchain committed:
Revert "Merge feat/managers into fix/v6-consolidation"

This reverts commit 99845f263bddcdd41edf2d9fb8f602e379b2c17d.

masonchain committed:
Merge branch 'feat/managers' into fix/v6-consolidation

masonchain committed:
fix: merge conflict

masonchain committed:
Merge branch 'feat/danger-zone' into fix/v6-consolidation

masonchain committed:
fix: infinite rerender bug

masonchain committed:
fix: manager table ready for decoded managers

masonchain committed:
fix: gas report

masonchain committed:
fix: strip out old error context

masonchain committed:
fix: badge token id routes

masonchain committed:
Merge branch 'fix/v6-consolidation' into feat/errors

masonchain committed:
fix: gas report

381e35 committed:
fix: web3 hex

nftchance committed:
fix: gas report

381e35 committed:
fix: bug fixes

381e35 committed:
fix: loose console log

nftchance committed:
fix: gas report

381e35 committed:
fix: low level bugs

381e35 committed:
Merge branch 'fix/v6-consolidation' of https://github.com/FlipsideCrypto/badger into fix/v6-consolidation

nftchance committed:
fix: gas report

masonchain committed:
fix: unnamed chain bug

masonchain committed:
fix: undefined modules bug

masonchain committed:
fix: isOwner undefined on creation

masonchain committed:
fix: removing managers functionality

masonchain committed:
fix: gas report

masonchain committed:
Merge branch 'fix/v6-consolidation' into feat/errors

381e35 committed:
feat: discover base

nftchance committed:
fix: gas report

381e35 committed:
fix: better action bar

381e35 committed:
Merge branch 'fix/v6-consolidation' of https://github.com/FlipsideCrypto/badger into fix/v6-consolidation

nftchance committed:
fix: gas report

381e35 committed:
fix: styling cleanup

381e35 committed:
Merge branch 'fix/v6-consolidation' of https://github.com/FlipsideCrypto/badger into fix/v6-consolidation

nftchance committed:
fix: gas report

381e35 committed:
fix: error component order

nftchance committed:
fix: gas report

381e35 committed:
feat: discovery foundation

nftchance committed:
fix: gas report

381e35 committed:
fix: disable slither

381e35 committed:
Merge branch 'feat/discovery' of https://github.com/FlipsideCrypto/badger into feat/discovery

nftchance committed:
fix: gas report

381e35 committed:
fix: wrapper & useTabActive

381e35 committed:
feat: address mirroring

Allows you to view the state of badger as another address.

This extends into the deepest layer before writes.

381e35 committed:
fix: socket flattening

nftchance committed:
fix: gas report

381e35 committed:
fix: background dataset loading

nftchance committed:
fix: gas report

381e35 committed:
fix: dashboard title

381e35 committed:
Merge branch 'feat/discovery' of https://github.com/FlipsideCrypto/badger into feat/discovery

nftchance committed:
fix: gas report

masonchain committed:
wip: pending transactions

masonchain committed:
fix: unused hook

381e35 committed:
fix: not found action bar

381e35 committed:
feat: functional discovery + org retrieveal

nftchance committed:
fix: gas report

381e35 committed:
feat: improve orgform performance

nftchance committed:
fix: gas report

381e35 committed:
feat: updating consumer organization pk

nftchance committed:
fix: gas report

381e35 committed:
fix: better dashboardloader

nftchance committed:
fix: gas report

381e35 committed:
fix: better useUser

nftchance committed:
fix: gas report

381e35 committed:
feat: object retrieval

nftchance committed:
fix: gas report

381e35 committed:
fix: general cleanup

nftchance committed:
fix: gas report

masonchain committed:
feat: transaction loading

381e35 committed:
feat: loaded form

masonchain committed:
Merge pull request #226 from FlipsideCrypto/feat/discovery

feat: organization discovery w/ user exploration

nftchance committed:
fix: gas report

masonchain committed:
fix: gas report

masonchain committed:
Merge branch 'fix/v6-consolidation' into feat/errors

381e35 committed:
fix: loadable badge form

381e35 committed:
fix: debounce badgeform

381e35 committed:
fix: loose improt

381e35 committed:
fix: deprecate hardhat workflow

381e35 committed:
fix: isEdit as managed

381e35 committed:
feat: go home on logout

381e35 committed:
fix: dump objects and socket on wallet change

381e35 committed:
fix: dashboardloader prep

381e35 committed:
fix: changing connected wallets (#231)

Implements a simple fix where the leading organizations from the socket are dumped when the connected wallet is changed. By doing it this way, all child architecture will be handled in line without needing to manage the state upwards.

381e35 committed:
feat: chain:ethereum_address pk retrieval

381e35 committed:
fix: loaded references

masonchain committed:
feat: window tree and transaction loading

381e35 committed:
fix: small prep

masonchain committed:
fix: structure

381e35 committed:
feat: badge archiving

381e35 committed:
fix: struct cleanup

381e35 committed:
fix: loader prep

381e35 committed:
feat: no more onchain archived

381e35 committed:
feat: no more onchain archived (#236)

381e35 committed:
feat: no more onchain symbol

381e35 committed:
fix: get name from uri

381e35 committed:
fix: get archived from ipfs with loader

381e35 committed:
feat: badge archiving (#235)

masonchain committed:
fix: ok new structure much better

masonchain committed:
fix: transaction animation in and out

masonchain committed:
fix: unused components

masonchain committed:
feat: loading and transaction messages

masonchain committed:
fix: gas report

masonchain committed:
fix: unused hook

masonchain committed:
Merge branch 'feat/errors' of https://github.com/FlipsideCrypto/badger into feat/errors

masonchain committed:
fix: gas report

masonchain committed:
fix: small holder table bug

masonchain committed:
Merge branch 'feat/errors' of https://github.com/FlipsideCrypto/badger into feat/errors

masonchain committed:
fix: gas report

masonchain committed:
fix: small style bug

381e35 committed:
Merge branch 'fix/deployment-cleanup' into feat/errors

381e35 committed:
fix: org design bug

381e35 committed:
feat: url encoded art links

381e35 committed:
Merge branch 'fix/deployment-cleanup' into feat/errors

381e35 committed:
feat: get symbol from uri

381e35 committed:
feat: uri as a database (#237)

Introduces the functionality that stores archived states in an object's metadata and tags in an organization's metadata rather than paying for that data onchain.

Accumulatively, this moves a significant amount of data offchain into normal metadata; still being tracked with the indexer as the URI update is always emitted for both Organizations and Badges.

381e35 committed:
Merge branch 'fix/deployment-cleanup' into feat/errors

masonchain committed:
fix: rerender loop

masonchain committed:
fix: click tracking for all tx events

381e35 committed:
feat: long org names

381e35 committed:
fix: filter errored organizations from discover

masonchain committed:
fix: not hiding dashboard loader

masonchain committed:
fix: table styling width

masonchain committed:
fix: style bugs

381e35 committed:
fix: logic correction

381e35 committed:
fix: self retrieved organizations (#245)

381e35 committed:
feat: card styling update

381e35 committed:
fix: org card layout

381e35 committed:
feat: final org cards

381e35 committed:
feat: better menu button

381e35 committed:
feat: action bar styling cleanup

381e35 committed:
fix: better org cards

381e35 committed:
fix: border radius

381e35 committed:
fix: badge form correction

381e35 committed:
fix: loose log

masonchain committed:
fix: loader messages

381e35 committed:
fix: image url logic

381e35 committed:
fix: uploaded images

381e35 committed:
fix: loose log

381e35 committed:
fix: editing images (#246)

masonchain committed:
fix: logout bug

381e35 committed:
feat: errors and transaction loading (#240)

381e35 committed:
fix: deployment cleanup + edge case testing (#229)

381e35 committed:
fix: v6 consolidation (#223)

sftchance committed:
fix: discovery cleanup

sftchance committed:
feat: chain icon display and address copy

sftchance committed:
feat: org card and manager table

sftchance committed:
feat: fix badge info display

381e35 committed:
feat: ux love (#249)

masonchain committed:
fix: new holder last login bug

masonchain committed:
fix: deleting managers functionality

masonchain committed:
fix: gettimesince util

masonchain committed:
fix: custom images

masonchain committed:
fix: proper custom image naming

masonchain committed:
fix: pfp undefined bug

masonchain committed:
fix: org discovery name undefined bug

masonchain committed:
fix: close navbar on click out

381e35 committed:
feat: landing page accordion images

381e35 committed:
fix: accordion bug created in faq

381e35 committed:
fix: logo in navbar now centered

381e35 committed:
feat: navbar link hover animations

381e35 committed:
Merge branch 'feat-badger-day' into fix/v6-consolidation

381e35 committed:
feat: responsive landing page accordion images

381e35 committed:
Merge branch 'fix/v6-consolidation' of https://github.com/FlipsideCrypto/badger into fix/v6-consolidation

381e35 committed:
feat: landing page accordion images

381e35 committed:
fix: documentation links & copyright

381e35 committed:
feat: add prettier

381e35 committed:
fix: update to walletconnect v2

381e35 committed:
feat: add safe support
